x-pack/plugins/reporting/public/share_context_menu/register_csv_reporting.tsx

```diff
@@ -82,33 +109,50 @@ export const reportingCsvShareProvider = ({
           disabled: licenseDisabled,
           ['data-test-subj']: 'CSVReports',
           sortOrder: 1,
+          onClick: openCsvModal,
         },
         panel: {
           id: 'csvReportingPanel',
           title: panelTitle,
-          content: (
-            <ReportingPanelContent
-              requiresSavedState={false}
-              apiClient={apiClient}
-              toasts={toasts}
-              uiSettings={uiSettings}
-              reportType={CSV_JOB_TYPE}
-              layoutId={undefined}
-              objectId={objectId}
-              getJobParams={getJobParams}
-              onClose={onClose}
-              theme={theme}
-            />
-          ),
+          content: openCsvModal,
         },
       });
     }
 
     return shareActions;
   };
 
+  const getJobsParams = ({ objectType, sharingData }: ShareContext) => {
+    const getSearchSource = sharingData.getSearchSource as ({
+      addGlobalTimeFilter,
+      absoluteTime,
+    }: {
+      addGlobalTimeFilter?: boolean;
+      absoluteTime?: boolean;
+    }) => SearchSourceFields;
+
+    const jobParams = {
+      title: sharingData.title as string,
+      objectType,
+      columns: sharingData.columns as string[] | undefined,
+    };
+
+    const getJobParams = (forShareUrl?: boolean) => {
+      return {
+        ...jobParams,
+        searchSource: getSearchSource({
+          addGlobalTimeFilter: true,
+          absoluteTime: !forShareUrl,
+        }),
+      };
+    };
+    console.log({ getJobParams });
+    return getJobParams;
+  };
+
   return {
     id: 'csvReports',
     getShareMenuItems,
+    jobProviderOptions: getJobsParams as unknown as JobParamsProviderOptions,
```

Comment:
Same as here - https://github.com/elastic/kibana/pull/171253/files#r1439918059 - let's try to do without this added field in `ShareMenuProvider`.